File: prepilot/view_prepilot.m
s0 = load('S0\S0_block1_PertrurbExpPilot.mat');
s1 = load('S1\S1_block1_PertrurbExpPilot.mat');

% tabular data only, not auditory
responses = [s0.whole_data; s1.whole_data];

% do we have the correct amount of data
height(responses)
height(responses(responses.ID == 1, :)) % participant 1
height(responses(responses.ID == 0, :)) % participant 0

% did the ost work?
height(responses(responses.OST_worked == 1, :))

% create a column that says if the response is correct or not
responses.correct = nan([height(responses) 1]);

idCorrect = find(responses.pert_magnitude > 0 & responses.updown_response == 1);
responses.correct(idCorrect) = true;

idCorrect = find(responses.pert_magnitude < 0 & responses.updown_response == 0);
responses.correct(idCorrect) = true;

idIncorrect = find(responses.pert_magnitude < 0 & responses.updown_response == 1);
responses.correct(idIncorrect) = false;

idIncorrect = find(responses.pert_magnitude > 0 & responses.updown_response == 0);
responses.correct(idIncorrect) = false;

% are the participants correct?
height(responses(responses.correct == 1, :))
height(responses(responses.correct == 0, :))

% correct responses by perturbation magnitude
perts = unique(responses.pert_magnitude);
perttype_agg = [];
for i = 1:length(perts)
    pert = perts(i);
    if (pert == 0)
        continue;
    end

    ids = find(responses.correct == 1 & responses.pert_magnitude == pert);
    perttype_agg = [perttype_agg; [pert length(ids)]];
end

% display correct reponses by pert. magnitude in bar plot
bar(perttype_agg(:, 1), perttype_agg(:, 2))

File: prepilot/view_prepilot.r
library(ggplot2)
library(dplyr)
library(tidyr)
library(janitor)

responses <- read.csv('responses.csv')

#do we have the correct numbers?
nrow(responses)
nrow(subset(responses, ID==1))
nrow(subset(responses, ID==0))

#did the ost work?
nrow(subset(responses, OST_worked==0))

#create column that says if the response is correct
responses$updown_correct <- NA
responses[responses$pert_magnitude > 0 & responses$updown_response == 1,]$updown_correct <- TRUE
responses[responses$pert_magnitude < 0 & responses$updown_response == 0,]$updown_correct <- TRUE
responses[responses$pert_magnitude > 0 & responses$updown_response == 0,]$updown_correct <- FALSE
responses[responses$pert_magnitude < 0 & responses$updown_response == 1,]$updown_correct <- FALSE

#are they correct?
nrow(subset(responses, updown_correct == TRUE))
nrow(subset(responses, updown_correct == FALSE))
# -> more correct than not

#correct responses by perturbation magnitude
updown_agg <- aggregate(updown_correct ~ pert_magnitude, data=responses, FUN="sum")
updown_agg$pert_magnitude <- factor(updown_agg$pert_magnitude)
levels(updown_agg$pert_magnitude)


ggplot(data=updown_agg, aes(x=pert_magnitude, y=updown_correct)) +
  geom_col(stat="identity") +
  ggtitle('Correct responses by magnitude of perturbation (max 40)') +
  xlab('magnitude of perturbation') +
  ylab('correct answers') +
  theme_classic()

#count confidence responses
by_confidence <- responses %>% group_by(how_noticeable_response)
by_confidence <- by_confidence %>% summarise(n = n())

ggplot(data=by_confidence, aes(x=how_noticeable_response, y=n)) +
  geom_col(stat="identity") +
  ggtitle('How likely are participants to answer 0,1,2 or 3') +
  xlab('confidence rating') +
  ylab('number of answers') +
  theme_classic()

#correct responses by confidence rating

tabyl(responses, how_noticeable_response, updown_correct)

tabyl(responses, how_noticeable_response, updown_correct) %>%
  adorn_percentages("row") %>%
  adorn_pct_formatting(digits = 1)

confidence_agg <- tabyl(responses, how_noticeable_response, updown_correct) %>% adorn_percentages("row")
confidence_agg <- rename(confidence_agg,  correct_responses = 'TRUE')
confidence_agg <- rename(confidence_agg,  incorrect_responses = 'FALSE')
confidence_agg <- rename(confidence_agg,  no_perturbation = NA_)

confidence_agg <- pivot_longer(confidence_agg, cols = c(correct_responses, incorrect_responses, no_perturbation), names_to='category')

ggplot(confidence_agg, aes(x=how_noticeable_response, y = value)) + 
  geom_bar(stat="identity", aes(fill=category)) +
  ggtitle('Percentage correct by confidence rating') +
  xlab('confidence rating') +
  ylab('percentage correct') +
  theme_classic()

